File: LeetCode/p452.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>> &points) {
        if (points.size() == 0)
            return 0;
        sort(points.begin(), points.end());
        int arrow = 1;
        int min_t = 0x7fffffff;
        for (auto & point : points) {
            auto &p = point;
            if (p[0] <= min_t) {
                min_t = min(min_t, p[1]);
            } else {
                ++arrow;
                min_t = point[1];
            }
        }
        return arrow;
    }
};

File: LeetCode/p452.rs
impl Solution {
  pub fn find_min_arrow_shots(mut points: Vec<Vec<i32>>) -> i32 {
    if points.is_empty() {
      return 0;
    }
    points.sort_unstable_by_key(|x| x[0]);
    let mut ret = 1;
    let mut bm = i32::MAX;
    for i in points.iter() {
      if i[0] > bm {
        ret += 1;
        bm = i[1];
      } else {
        bm = std::cmp::min(bm, i[1]);
      }
    }
    ret
  }
}

